tests/zfs-tests/include/libtest.shlib

```diff
@@ -2483,6 +2518,113 @@ function is_physical_device #device
 }
 
 #
+# Check if the given device is a real device (ie SCSI device)
+#
+function is_real_device #disk
+{
+	typeset disk=$1
+	[[ -z $disk ]] && log_fail "No argument for disk given."
+
+       if is_linux; then
+		$LSBLK -S $DEV_RDSKDIR/$disk > /dev/null 2>&1
```

Comment:
Using `lsblk -S` to do this check is going to result in virtual block devices being excluded.  Ideally we want the option of running all the tests against virtblk (kvm) and xvd (xen) virtual block devices for automated testing.  This check probably excludes hd* style devices too, but I didn't verify that.  Let's do this in a way which allows everything which provides a full block device interface.